client/src/user-context.tsx

```diff
@@ -236,10 +238,10 @@ export function UserDataProvider(props: { children: React.ReactNode }) {
     }
   }, [data]);
 
-  let userData = data || getSessionStorageData();
+  const userData = isLoading ? getSessionStorageData() : error ? null : data;
```

Comment:
Brain's a bit slow today, perhaps this is easier to read?

```suggestion
  const userData = data || error ? null : getSessionStorageData();
```

Reply:
I'm not sure, I tend to say it's more confusing. How about:

```suggestion
  let userData;
  if (isLoading) {
    userData = getSessionStorageData();
  } else if (error) {
    userData = null;
  } else {
    userData = data;
  }
```

Then again, I wonder if `userData` is `null` if `error` anyways?

Reply:
Nah, stick with the ternaries, let's keep `userData` a `const` :)